yellowbrick/features/rfecv.py

```diff
@@ -0,0 +1,331 @@
+
+    cv_scores_ : array of shape [n_subsets_of_features, n_splits]
+        The cross-validation scores for each subset of features and splits in
+        the cross-validation strategy.
+
+    rfe_estimator_ : sklearn.feature_selection.RFE
+        A fitted RFE estimator wrapping the original estimator. All estimator
+        functions such as ``predict()`` and ``score()`` are passed through to
+        this estimator (it rewraps the original model).
+
+    Notes
+    -----
+    This model wraps ``sklearn.feature_selection.RFE`` and not
+    ``sklearn.feature_selection.RFECV`` because access to the internals of the
+    CV and RFE estimators is required for the visualization. The visualizer
+    does take similar arguments, however it does not expose the same internal
+    attributes.
+
+    Additionally, the RFE model can be accessed via the ``rfe_estimator_``
+    attribute. Once fitted, the visualizer acts as a wrapper for this
+    estimator and not for the original model passed to the model. This way the
+    visualizer model can be used to make predictions.
+
+    .. caution:: This visualizer requires a model that has either a ``coef_``
+        or ``feature_importances_`` attribute when fitted.
+    """
+
+    def __init__(self, model, ax=None, step=1, groups=None, cv=None,
+                 scoring=None, **kwargs):
+
+        # Initialize the model visualizer
+        super(RFECV, self).__init__(model, ax=ax, **kwargs)
+
+        # Set parameters
+        self.set_params(step=step, groups=groups, cv=cv, scoring=scoring)
+
+    def fit(self, X, y=None):
+        """
+        Fits the RFECV with the wrapped model to the specified data and draws
+        the rfecv curve with the optimal number of features found.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples) or (n_samples, n_features), optional
+            Target relative to X for classification or regression.
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the RFECV visualizer.
+        """
+        X, y = check_X_y(X, y, "csr")
+        n_features = X.shape[1]
+
+        # This check is kind of unnecessary since RFE will do it, but it's
+        # nice to get it out of the way ASAP and raise a meaningful error.
+        if 0.0 < self.step < 1.0:
+            step = int(max(1, self.step * n_features))
+        else:
+            step = int(self.step)
+
+        if step < 0:
+            raise YellowbrickValueError("step must be >0")
+
+        # Create the RFE model
+        rfe = RFE(self.estimator, step=step)
+        n_feature_subsets = np.arange(1, n_features+1)
+
+        # Create the cross validation params
+        # TODO: handle random state
```

Comment:
this seems like a TODO that probably be in here, could you add this into this PR?

Reply:
I can take a look at it, the issue is that the mechanism I'm using, [`cross_val_score`](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html), doesn't take `random_state` as a parameter. 

In the `DescriminationThreshold` visualizer, you implemented `random_state` which was passed to `train_test_split`, later I updated it to take a `cv` argument, which in order to use the random state, I had to implement a [`_check_cv`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/threshold.py#L416) method that mirrored your use of `train_test_split`. This was loosely based off of [`sklearn.model_selection.check_cv`](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/model_selection/_split.py#L1862), though the sklearn mechanism is much more robust than what I did and I'm not sure if I like how I went about things in threshold.py.

So, the reason I added it as a TODO is because I'm not sure what to do here. The only option to generically handle the `cv` argument with a random state is to create our own splitter, it seems like a shame to implement this just for that purpose since it will be rigid with respect to skearn's CV system. I assume the reason `cross_val_score` doesn't take a random_state parameter is that you can simply pass it a splitter with the random state already set (which is what I'm doing in the tests). 

Any thoughts on this?

Reply:
So on further investigation, it actually looks like no random state is needed:

https://gist.github.com/bbengfort/ae9f9a52ebb55033845553722175c405

The cv method is deterministic unless `shuffle=True`, which is always false when you pass in `None` or an integer. Does this look right?